src/server/qgsconfigcache.h

```diff
@@ -30,20 +31,59 @@
 #include "qgsproject.h"
 #include "qgsserversettings.h"
 
+#ifndef SIP_RUN
+
+
+#endif
+
+class QgsConfigCache;
+
+/**
+ * \ingroup server
+ * \brief Abstract cache strategy for server configuration
+ * \since QGIS 3.24
+ */
+class SERVER_EXPORT QgsAbstractCacheStrategy  : public QObject
+{
+    friend class QgsConfigCache;
```

Comment:
No reason to use all the friend magic. Just make this public api. And make an internal private virtual to be implemented by subclasses.

Reply:
In the actual implementation, the strategy si responsible for connecting `QgsConfigCache` private slots methods so these must be made public if we do not use  `friends` declaration.

Reply:
Do they need to be private?